### PR TITLE
Various minor improvements

### PR DESCRIPTION
**Pull request by candleindark**
This PR brings in the following minor improvements to #308:

1. Improvement in the last example for the query syntax
2. Replace a use of list with use of a generator for performance.
3. Update to the modern SQLAlchemy syntax in `overview.py` and `test_search.py`.

### CODE REVIEW

File: datalad_registry/templates/overview.html [hunk 1]
```diff
@@ -106,7 +106,10 @@ <h2>Search query syntax</h2>
               <span class="query">(url:github.com OR url:gin.g-node) AND (Haxby (James OR Jim))</span>.
             </li>
             <li>You can restrict a search to specific metadata extractors by specifying one or more of them,
-              e.g., <span class="query">metadata[bids_dataset,metalad_core]:"BIDSVersion\": \"v"</span></li>
+              e.g., <span class="query">metadata[bids_dataset,metalad_core]:"BIDSVersion\":\"v"</span>
```
[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad_registry/templates/overview.html:109]: we definitely need a test case... it is actually quite confusing now:

if I download metadata record, e.g. https://registry.datalad.org/api/v2/url-metadata/22448 I see that there is no spaces:

image

but when I search (with our simple search ATM), I get no results if I search for a string without space: https://registry.datalad.org/overview/?filter=%22%40id%22%3A%22d376&sort=update-desc and get results if with a single space: https://registry.datalad.org/overview/?filter=%22%40id%22%3A+%22d376&sort=update-desc . So I guess whatever transformation we do for JSONB while searching - it adds formatting, but doesn't add it while downloading.  Ideally we should have it uniform one way or another

[the remark this answers is not in the document]
Comment (yarikoptic) [reply datalad_registry/templates/overview.html:109]: FWIW -- I pushed into my PR that added test case: indeed need a space there!  I even tested it rebased on master (presumably with JSONB?) -- still passed with the space. So let's bring space back here and below in a comment, and rebase on top of current enh-search.

note:  I also renamed that test into `test_search_cases`.

File: datalad_registry/tests/test_search.py [hunk 2]
```diff
@@ -14,7 +15,7 @@
         # of them, but let's test that error message as expected
         ("ds_id:=example", VisitError, "Operation = is not implemented"),
         # r'(haxby or halchenko) AND metadata:BIDSmetadata[bids_dataset,metalad_core]:'
-        # r'"BIDSVersion\": \"v"',
+        # r'"BIDSVersion\":\"v"',
```
Comment (yarikoptic) [inline datalad_registry/tests/test_search.py:18]:
```suggestion
        # r'"BIDSVersion\": \"v"',
```

File: datalad_registry/templates/overview.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (candleindark) [reply datalad_registry/templates/overview.html]: That's very interesting. AFAIK, even with the JSONB type enabled, our current arrangement searches the text form of the data not its JSONB form. Sure, let's keep it this way.

**Pull request merged**